### DISCUSSION TITLE
import HERBS probe data in alignment gui

### DISCUSSION THREAD

**Discussion by Andrianarivelo** [AllenNeuralDynamics/ibl-ephys-alignment-gui#3] [Q&A]
Hi, 
I exported NPX 2.0 probes data (as pkl files, one for each shank), how can I format them to a format compatible with the ephys alignment-gui ? 
Thanks

**Comment by jsiegle** [comment]
This fork of the IBL GUI uses the same input format as the original, which is documented here: https://github.com/int-brain-lab/iblapps/wiki/4.-Preparing-data-for-ephys-GUI

We have created a script for exporting ephys data into the compatible format when starting from a SpikeInterface sorting/recording objects, which you may find helpful: https://github.com/AllenNeuralDynamics/aind-ephys-ibl-alignment-gui-converter/blob/main/code/run_capsule.py

**Comment by Aexolowski** [comment]
I have a similar question, but want to know how to format the histological data from HERBS (the pickle files, one for each shank) to be able to use them with the IBL ephys alignment tool.

**Comment by jsiegle** [comment]
This is probably a better question to ask in the HERBS repo. Hopefully it's not too difficult to figure out how to .pkl files to the IBL-GUI-specific JSONs.

**Reply by Aexolowski** [reply]
Sadly the herbs forum is very quiet lately, I have tried to ask them also a while ago but did not get any replies. We do have the XYZ coordinates for each shank, but we are unsure about the transformation from Allen CCF into IBL space, as we don't know if we have to use the mlapdv or the apdvml order...

**Reply by jsiegle** [reply]
The conventions are documented here: https://github.com/int-brain-lab/iblapps/wiki/4.-Preparing-data-for-ephys-GUI#coordinate-systems

It's x/y/z = ML/AP/DV, and all of the coordinates are relative to the bregma point (located at ML = 5739 um, AP = 5400 um and DV = 332 um)

**Reply by Ic3-Fury** [reply]
Hello @jsiegle, I work with @Aexolowski and had a question. I have created a .json file with the relevant coordinates that were transformed from HERBS but end up getting an error when I try to import the data from the IBL alignment GUI. I've attached screenshots of the error message and the contents of the alf folder if that helps. I looked at the sample data folder given by IBL as well and my .json file labelled xyz_picks looks the exact same structurally. Any help here would be much appreciated!
 
![Screenshot 2025-06-26 172120](https://github.com/user-attachments/assets/e09156bc-26ef-4d33-9899-407fe9890729)

![image](https://github.com/user-attachments/assets/6ccb3d96-09c6-4774-8bbc-9829e3cff5f7)

**Reply by jsiegle** [reply]
Can you attach xyz_picks.json file here?

**Reply by Ic3-Fury** [reply]
[xyz_picks.json](https://github.com/user-attachments/files/20955746/xyz_picks.json)
Here is the xyz_picks file...we are still working on converting the coordinates to the IBL Bregma space using the code found here: https://github.com/int-brain-lab/iblapps/wiki/4.-Preparing-data-for-ephys-GUI. The points we have here are in the Allen CCF space. We appreciate the help!

**Reply by Ic3-Fury** [reply]
Following up, we ran the code on our xyz_picks file to convert from CCF (mlapdv) --> Bregma (mlapdv) and it output a very small number - in the thousandths or the ten thousandths decimal place ([xyz_picks_bregma.json](https://github.com/user-attachments/files/20956736/xyz_picks_bregma.json)). When comparing it to the sample xyz_picks file provided by IBL ([xyz_picks.json](https://github.com/user-attachments/files/20956745/xyz_picks.json)) we see that the numbers in this json file are in the thousands, not in decimals. 

Furthermore, we ran the sample code as is listed on the page:

`from iblatlas import atlas
import numpy as np

res = 25
brain_atlas = atlas.AllenAtlas(res)

ccf_mlapdv = np.array([[3000, 4000, 3000], [6000, 6000, 500] ], dtype=float)
bregma_mlapdv = brain_atlas.ccf2xyz(ccf_mlapdv, ccf_order='mlapdv')`

And we received the output coordinates: [-0.002739,  0.0014  , -0.002668], [ 0.000261, -0.0006  , -0.000168]. So we are unsure how the xyz_picks sample file relates to these decimal values.

**Reply by jsiegle** [reply]
I believe the `AssertionError` is due to a problem with the file naming, as it's not finding exactly one `xyz_picks.json` file in the data directory. Based on the screenshot, it may be because the `.json` extension is missing.

The xyz_picks file should be in units of microns, so the original scaling looks correct.

Looping in @mayofaulkner, who may be better suited to address some of these questions.

**Reply by Ic3-Fury** [reply]
Hello! Really appreciate you looping in Mayo, I think that would be great help here. We already tried renaming file "xyz_picks.json" and checked that it was a .json in it's properties...same error.

**Reply by mayofaulkner** [reply]
Hello, 

The output from the ccf2xyz conversion is in metres but you will need to convert to um ( x 1e6) and then write to the json file. This should explain the difference in scales. 

Is this single shank or multi shank data? If it is single shank then the file match is '*xyz_picks.json', if multi shank then you will need to provide one xyz_picks per shank and name them e.g `*xyz_picks_shank1.json` `*xyz_picks_shank2.json` etc

**Reply by Ic3-Fury** [reply]
Hi Mayo, the numbers now make a lot more sense knowing that they're in meters. We are using NPX 2.0 probes, so I'll work on making the files for each shank and naming them as you have suggested. I'll get back to you here as soon as I finish that and seeing if it works! Hopefully this should resolve the issues we have been having. Again, we really appreciate both yours and Josh's time and help!
